src/BugsnagUnity/FileManager.cs

```diff
@@ -246,74 +234,40 @@ internal static void RemoveCachedSession(string id)
             }
         }
 
-        private static void RemovePendingSession(string id)
+        internal static string[] GetCachedPayloadPaths()
         {
-            _pendingSessions.RemoveAll(item => item.Id == id);
+            var cachedPayloadPaths = new List<string>();
+            cachedPayloadPaths.AddRange(_cachedSessions);
+            cachedPayloadPaths.AddRange(_cachedEvents);
+            return cachedPayloadPaths.ToArray();
         }
 
-        internal static List<IPayload> GetCachedPayloads()
+        internal static IPayload GetPayloadFromCachePath(string path)
         {
-            var cachedPayloads = new List<IPayload>();
-
-            //Get cached sessions
-            foreach (var cachedSessionPath in _cachedSessions)
-            {
-                var json = File.ReadAllText(cachedSessionPath);
-                var sessionReportFromCachedPayload = new SessionReport(_configuration, ((JsonObject)SimpleJson.DeserializeObject(json)).GetDictionary());
-                cachedPayloads.Add(sessionReportFromCachedPayload);
-            }
-
-            //get cached events and run them through on send callbacks
-            var cachedEvents = _cachedEvents.ToList();
-            foreach (var cachedEventPath in cachedEvents)
+            if (File.Exists(path))
             {
-                var json = File.ReadAllText(cachedEventPath);
-                var eventReportFromCachedPayload = new Report(_configuration, ((JsonObject)SimpleJson.DeserializeObject(json)).GetDictionary());
-                var shouldDiscard = false;
-                foreach (var onSendErrorCallback in _configuration.GetOnSendErrorCallbacks())
-                {
-                    try
-                    {
-                        if (!onSendErrorCallback.Invoke(eventReportFromCachedPayload.Event))
-                        {
-                            shouldDiscard = true;
-                            break;
-                        }
-                    }
-                    catch
-                    {
-                        // If the callback causes an exception, ignore it and execute the next one
-                    }
-                }
-                if (shouldDiscard)
+                if (path.Contains("session"))
```

Comment:
Suggest making a `const` out of ".session" and ".event" and testing against that rather than string values to make it clear what you're looking for. Presumably this could also be an `EndsWith`?

Reply:
done 👍